### PR TITLE
Select Pedals

### PR DESCRIPTION
**Pull request by hodasemi**
Hi,

as follow up from the discussion in #89 I started working on a solution. I added a ComboBox above the combine pedals area to select from which source the input should come from.

grafik
grafik

There will always be the 'Default' option, because I guess there is no way to check if pedals are actually connected to the wheel base. The current work only adds to the UI, queries possible devices and adds it to the ComboBox. I think the missing parts are

* [x] use the selected pedal as input source and display its use below
* [x] store the pedal selection in the profile

--- 

~~The ultimate goal should be~~

~~* [ ] create virtual device to combine the source devices~~

### CODE REVIEW

**Comment by irseny** [comment]
Would the combined virtual device be compatible with force feedback though?
I do not know the details in linux so I might be wrong about this:
I think the virtual device would have to advertise FFB capabilities and also forward the game's FFB signals to the wheel device?

**Comment by hodasemi** [comment]
I guess so. I don't know these things either, but that information must be out there somewhere ...
I would at least try to solve the problem.

**Comment by berarma** [comment]
I think this PR should be about adding multi-device support without creating any virtual device. That should be left for another PR in case anyone wants to do the work. Although I still think it would be better to do that work towards a more generic app.

I would like the devices to be shown together, the pedal selection combo directly below the wheel. Later, more devices could be added like shifter and handbrake.

I can't review the code now but will do. Feel free to keep working on it in the mean time.

**Comment by hodasemi** [comment]
The core features are now in place. You can now review the code.

This is how it looks:
grafik

Currently you can only view the behavior inside oversteer. There is no virtual device that routs input into a single device. I agree that it would be better to make it more general, that the users can map the axes they need to the correct one, instead of hard coding them for each and every device. 
Sadly this still doesn't solve the core issue I tried to solve. I will take a deeper look into virtual devices, especially routing force feedback back to the physical device.

File: data/udev/99-thrustmaster-pedal-perms.rules [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Match kernel name of device, rather than ATTRS{idProduct} and ATTRS{idVendor}
```
Comment (berarma) [inline data/udev/99-thrustmaster-pedal-perms.rules:1]: I'd rather put it in the thrustmaster udev file despite having "wheel" in its name. I don't feel like creating many more udev files.

**Review by berarma — CHANGES_REQUESTED** [review]
It looks pretty good. There's some things I'd change before merging that I've commented in the reviewed code. And then, I'm not sure about the way to handle multi-devices, I think it should be more flexible for accepting more devices later, but that can be worked after merging into a custom development branch.

Why is `update_pedal_list` so different to `update_device_list`? I hoped it would be almost the same code but using different id lists.

**Comment by hodasemi** [comment]
> Why is `update_pedal_list` so different to `update_device_list`? I hoped it would be almost the same code but using different id lists.

This is what I also thought. The problem is that the entry that has a `ID_FOR_SEAT` does not have `DEVNAME`. Thus, I couldn't use the created entry for like anything at all. I decided to sort by usb-id and puzzle the required information together.

File: oversteer/gui.py [hunk 2]
```diff
@@ -437,11 +474,40 @@ def process_events(self, events):
...
+                rdescriptors.append(input_pedals.fd)
+
+        r, _, _ = select.select(rdescriptors, [], [], timeout)
+
+        events = []
+
+        if input_device.fd in r:
+            events = list(self.device.read_events(input_device))
+
+        if self.pedals:
```
[the remark this answers is not in the document]
Comment (hodasemi) [reply oversteer/gui.py:480]: The filtering of the pedals is done in the exact next line if there is pedal present in `filter_pedal_input`.

**Review by berarma — APPROVED** [review]

**Comment by berarma** [comment]
Thanks! I'll shortly merge this in a new branch to test it more and maybe ready it to add more devices, like shifters and handbrakes. This could lead to more cool things.

**Comment by berarma** [comment]
I've created a branch named `multidevice` with these changes to work towards a multidevice app.

**Pull request closed without merging**